deploy/templates/nstemplatetiers/advanced/ns_code.yaml

```diff
@@ -122,6 +124,34 @@ objects:
     podSelector: {}
     policyTypes:
       - Ingress
+- apiVersion: networking.k8s.io/v1
+  kind: NetworkPolicy
+  metadata:
+    name: allow-from-dev-namespace
+    namespace: ${USERNAME}-code
+  spec:
+    ingress:
+      - from:
+        - namespaceSelector:
+            matchLabels:
+              name: ${USERNAME}-dev
+    podSelector: {}
+    policyTypes:
+      - Ingress
+- apiVersion: networking.k8s.io/v1
```

Comment:
Does it have to be a separate policy (one policy per namespace)? Or it's possible to define one policy which would accept traffic from namespaces with either one label **or** another? Basically is it possible to have some kind of **or** operation in the `- from`?

Reply:
I don't see way to do that. https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#networkpolicyingressrule-v1-networking-k8s-io

Even the label selector only supports ANDing things together: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#labelselector-v1-meta

Reply:
Hm..
> from | List of sources which should be able to access the pods selected for this rule. Items in this list are combined using a logical OR operation. If this field is empty or missing, this rule matches all sources (traffic not restricted by source). If this field is present and contains at least one item, this rule allows traffic only if the traffic matches at least one item in the from list.

Reply:
So sounds like this is what we need:
```
  spec:
    ingress:
      - from:
        - namespaceSelector:
            matchLabels:
              name: ${USERNAME}-dev
        - namespaceSelector:
            matchLabels:
              name: ${USERNAME}-stage
```
right?

Reply:
Oops! Yes you're right. I misread it. I'll make the change.

Reply:
Done in https://github.com/codeready-toolchain/host-operator/pull/358/commits/34599fc68836ec12a22b9887db21bf808a7e0423